tests/src/main/java/net/neoforged/neoforge/debug/entity/EntityEventTests.java

```diff
@@ -75,6 +80,24 @@ static void entityAttributeModificationEvent(final DynamicTest test, final Regis
                 .thenSucceed());
     }
 
+    @GameTest
+    @EmptyTemplate
+    @TestHolder(description = "Tests if EntityInvulnerabilityCheckEvent prevents damage when modified.")
+    static void entityInvulnerabilityCheckEvent(final DynamicTest test, final RegistrationHelper reg) {
+        test.eventListeners().forge().addListener((final EntityInvulnerabilityCheckEvent event) -> {
+            if (event.getEntity() instanceof GameTestPlayer entity)
```

Comment:
"wildcard" tests like these are dangerous, please add a marker to distinguish the players. Maybe change its name.